util.go

```diff
@@ -75,13 +75,23 @@ func isPointerToStruct(v interface{}) (bool, error) {
 	return true, nil
 }
 
-func isPointerToSliceStructs(v interface{}) (bool, error) {
+func isPointerToStructOrMap(v interface{}) (bool, error) {
+	if el, err := ptrType(v); err != nil {
+		return false, err
+	} else if el.Kind() != reflect.Struct && el.Kind() != reflect.Map {
+		return false, errors.New("not a pointer to a struct or map")
+	}
+
+	return true, nil
+}
+
+func isPointerToSliceStructsOrMaps(v interface{}) (bool, error) {
 	if el, err := ptrType(v); err != nil {
 		return false, err
 	} else if el.Kind() != reflect.Slice {
 		return false, errors.New("not a pointer to a slice")
-	} else if el.Elem().Kind() != reflect.Struct {
-		return false, errors.New("not a pointer to a slice of structs")
+	} else if el.Elem().Kind() != reflect.Struct && el.Elem().Kind() != reflect.Map {
```

Comment:
Since we check twice would be nicer to declare the element kind, e.g. `else if k := el.Elem().Kind(); k != reflect.Struct && k != reflect.Map {`